spec/02-integration/05-proxy/13-server_tokens_spec.lua

```diff
@@ -160,6 +173,12 @@ describe("headers [#" .. strategy .. "]", function()
 
       teardown(helpers.stop_kong)
 
+      it("intermediate kong conf in prefix should contain 'headers=server_tokens'", function()
+        local conf = helpers.get_running_conf()
+        assert.equal("server_tokens", conf.headers[1])
+        assert.equal(1, #conf.headers)
+      end)
```

Comment:
imho, those tests are too implementation-specific - only the unit tests and a couple of integration tests are aware of the `.kong_env` file.

Instead, it might be better to write an integration test here, which will both be more resilient to implementation changes, and explicitly states the issue (which is nice for the future selves who will browse this test suite and modify this trying not to break it).

Such a test could be written if we load the `headers` value from a configuration file (instead of environment variables). We also try to keep track of regression test with a comment explaining their existence, which will make this test's importance clearer to the reader:
```lua
      it("can also be configured via configuration file", function()
        -- a regression test added with https://github.com/Kong/kong/pull/3419
        -- to ensure that the `headers` configuration value can be specified
        -- via config file as well as via environment variables (all
        -- other tests in this file only test the latter).
      end)
```

You may create a new configuration file under `spec/fixtures/headers.conf` which contains something like:
```
headers = server_tokens
```

And start Kong with:
```lua
          assert(helpers.kong_exec("restart -c spec/fixtures/headers.conf", {
            database = strategy,
            proxy_listen = "0.0.0.0:9000, 0.0.0.0:9443 ssl", -- a bit ugly but already done in `core_entities_invalidations_spec.lua`
          }))
```

Another pattern could be to duplicate `spec/kong_tests.conf` and append a `headers = ` directive.

This approach also has the benefit of not needed to expose `get_running_conf()`. I am also sceptical on testing the same thing 3 times, since a breaking change in the code will trigger 3 test failures instead of a clearly defined and documented one. This philosophy behind our tests are made our lives easier at times (and harder when it wasn't followed we well...).